// product-list/lib/components/guest-list.js
"use strict";

exports.__esModule = true;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _guestList = require("../../src/styles/guest-list.css");

var _guestList2 = _interopRequireDefault(_guestList);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _ref2 = _react2.default.createElement(
  "h1",
  null,
  "Guest List:"
);

var GuestList = function GuestList(_ref) {
  var invitees = _ref.invitees,
      toggleGuest = _ref.toggleGuest;


  var renderInvitees = function renderInvitees(inviteesArr) {
    return inviteesArr.map(function (invitee) {
      return _react2.default.createElement(
        "div",
        { key: invitee.name, className: _guestList2.default.guestName },
        _react2.default.createElement("input", {
          id: invitee.name,
          type: "checkbox",
          checked: invitee.invited,
          onChange: function onChange() {
            return toggleGuest(invitee);
          } }),
        _react2.default.createElement(
          "label",
          { htmlFor: invitee.name },
          invitee.name
        )
      );
    });
  };

  return _react2.default.createElement(
    "div",
    { className: _guestList2.default.guestList },
    _ref2,
    renderInvitees(invitees)
  );
};

GuestList.propTypes = {
  invitees: _react2.default.PropTypes.array,
  toggleGuest: _react2.default.PropTypes.func
};

exports.default = GuestList;
