// product-list/lib/components/product-list.js
"use strict";

exports.__esModule = true;

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _productList = require("../../src/styles/product-list.css");

var _productList2 = _interopRequireDefault(_productList);

var _renderFriend = require("./render-friend");

var _renderFriend2 = _interopRequireDefault(_renderFriend);

var _guestList = require("./guest-list");

var _guestList2 = _interopRequireDefault(_guestList);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var YourAwesomeComponent = function (_Component) {
  (0, _inherits3.default)(YourAwesomeComponent, _Component);

  function YourAwesomeComponent(props) {
    (0, _classCallCheck3.default)(this, YourAwesomeComponent);
    return (0, _possibleConstructorReturn3.default)(this, _Component.call(this, props));
  }

  YourAwesomeComponent.prototype.renderFriends = function renderFriends(friends, party) {
    var invitees = this.props.invitees;
    var partyTime = party ? _productList2.default.party : "";

    return friends.filter(function (friend) {
      return !!invitees.filter(function (invitee) {
        return invitee.name === friend.name && invitee.invited;
      }).length;
    }).map(function (friend) {
      return _react2.default.createElement(_renderFriend2.default, { className: partyTime, key: friend.name, friend: friend });
    });
  };

  YourAwesomeComponent.prototype.viewState = function viewState(view) {
    if (view) {
      return view;
    }
    return {
      intro: true,
      invite: true
    };
  };

  YourAwesomeComponent.prototype.houseParty = function houseParty(invitees, party) {
    return party ? _productList2.default.yourAwesomeComponent + " " + _productList2.default.house : _productList2.default.house;
  };

  YourAwesomeComponent.prototype.render = function render() {
    var _props = this.props,
        ourFriends = _props.ourFriends,
        invitees = _props.invitees,
        view = _props.view,
        message = _props.message,
        _toggleGuest = _props.toggleGuest;

    var party = invitees.length === invitees.filter(function (invitee) {
      return invitee.invited;
    }).length && invitees.length > 0;

    var _viewState = this.viewState(view),
        invite = _viewState.invite,
        intro = _viewState.intro;

    var houseParty = this.houseParty(invitees, party);

    return _react2.default.createElement(
      "div",
      null,
      invite && invitees.length > 0 && _react2.default.createElement(_guestList2.default, { invitees: invitees, toggleGuest: function toggleGuest(invitee) {
          return _toggleGuest(invitee);
        } }),
      _react2.default.createElement(
        "div",
        { className: _productList2.default.container },
        intro && !invitees.filter(function (invitee) {
          return invitee.invited;
        }).length && message(_productList2.default.message),
        _react2.default.createElement(
          "div",
          { className: houseParty },
          _react2.default.createElement(
            "div",
            { className: _productList2.default.room },
            this.renderFriends(ourFriends, party)
          )
        )
      )
    );
  };

  return YourAwesomeComponent;
}(_react.Component);

exports.default = YourAwesomeComponent;


YourAwesomeComponent.displayName = "YourAwesomeComponent";

YourAwesomeComponent.propTypes = {
  ourFriends: _react.PropTypes.array,
  message: _react.PropTypes.func,
  invitees: _react.PropTypes.array,
  view: _react.PropTypes.object,
  toggleGuest: _react.PropTypes.func
};

YourAwesomeComponent.defaultProps = {
  ourFriends: [],
  message: function message() {
    return "<p>Let's party! Un-comment the all the commented-out lines in the\n      playground then check the boxes on the GuestList to invite our friends to the party!</p>";
  },
  invitees: []
};
